### PR TITLE
Feature/simple avatar

### PR DESCRIPTION
**Pull request by YasserNezzari**

### CODE REVIEW

File: UltraleapTracking.uplugin [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
 	"FileVersion": 3,
-	"Version": 12,
-	"VersionName": "4.6.0",
+	"Version": 14,
+	"VersionName": "4.6.2",
```
Comment (MaxPalmer-UH) [inline UltraleapTracking.uplugin:4]: Should this now be 4.7.0?

File: Source/UltraleapTrackingCore/Private/OpenXRToLeapWrapper.cpp [hunk 2]
```diff
@@ -15,7 +15,7 @@
 #include "LeapUtility.h"
```
Comment (YasserNezzari) [inline Source/UltraleapTrackingCore/Private/OpenXRToLeapWrapper.cpp:15]: is this file supposed to be here ? may need to remove (looks like openxr)

File: Source/ThirdParty/BodyState/Private/AnimNode_ModifyBodyStateMappedBones.cpp [hunk 3]
```diff
@@ -65,61 +65,61 @@ void FAnimNode_ModifyBodyStateMappedBones::ApplyTranslation(const FCachedBoneLin
 	}
 
 	// is it the root?
-	if (&MappedBoneAnimData.CachedBoneList[0] == &CachedBone && WristBoneIndex > -1 && WristCachedBone &&
+	if (&MappedBoneAnimDataIn.CachedBoneList[0] == &CachedBone && WristBoneIndex > -1 && WristCachedBone &&
```
Comment (YasserNezzari) [inline Source/ThirdParty/BodyState/Private/AnimNode_ModifyBodyStateMappedBones.cpp:68]: check if index 0 exists in CachedBoneList[0]

File: Source/ThirdParty/BodyState/Private/AnimNode_ModifyBodyStateMappedBones.cpp [hunk 4]
```diff
@@ -65,61 +65,61 @@ void FAnimNode_ModifyBodyStateMappedBones::ApplyTranslation(const FCachedBoneLin
...
 void FAnimNode_ModifyBodyStateMappedBones::ApplyAutoCorrectRotation(const FTransform& WristBeforeMapping,
-	const FTransform& LeapWrist, FTransform& NewBoneTM, const FMappedBoneAnimData& MappedBoneAnimData)
+	const FTransform& LeapWrist, FTransform& NewBoneTM, const FMappedBoneAnimData& MappedBoneAnimDataIn)
 {
 	FQuat BoneQuat = NewBoneTM.GetRotation();
-	BoneQuat = MappedBoneAnimData.AutoCorrectRotation * BoneQuat;
-	BoneQuat = (BoneQuat * MappedBoneAnimData.OffsetTransform.GetRotation());
+	BoneQuat = MappedBoneAnimDataIn.AutoCorrectRotation * BoneQuat;
+	BoneQuat = (BoneQuat * MappedBoneAnimDataIn.OffsetTransform.GetRotation());
 	NewBoneTM.SetRotation(BoneQuat);
```
Comment (YasserNezzari) [inline Source/ThirdParty/BodyState/Private/AnimNode_ModifyBodyStateMappedBones.cpp:123]: check for nullptr "BoneQuat"

File: Source/ThirdParty/BodyState/Private/AnimNode_ModifyBodyStateMappedBones.cpp [hunk 5]
```diff
@@ -131,9 +131,9 @@ FVector CalculateAxis(const FTransform& Transform, const FVector& Direction)
 	return BoneForward;
 }
 void FAnimNode_ModifyBodyStateMappedBones::ApplyScale(const FCachedBoneLink& CachedBone, const FCachedBoneLink* CachedPrevBone,
-	FTransform& NewBoneTM, FTransform& PrevBoneTM, const FMappedBoneAnimData& MappedBoneAnimData)
+	FTransform& NewBoneTM, FTransform& PrevBoneTM, const FMappedBoneAnimData& MappedBoneAnimDataIn)
 {
-	if (!BSAnimInstance->ScaleModelToTrackingData || !MappedBoneAnimData.FingerTipLengths.Num())
+	if (!BSAnimInstance->ScaleModelToTrackingData || !MappedBoneAnimDataIn.FingerTipLengths.Num())
```
Comment (YasserNezzari) [inline Source/ThirdParty/BodyState/Private/AnimNode_ModifyBodyStateMappedBones.cpp:136]: check "BSAnimInstance" for nullptr

File: Source/UltraleapTrackingCore/UltraleapTracking.Build.cs [hunk 6]
```diff
@@ -102,7 +102,8 @@ public UltraleapTracking(ReadOnlyTargetRules Target) : base(Target)
 			PublicDependencyModuleNames.AddRange(
 				new string[]
 				{
-					"Engine",
+                    "ApplicationCore",
```
Comment (YasserNezzari) [inline Source/UltraleapTrackingCore/UltraleapTracking.Build.cs:105]: adjust format

File: Source/UltraleapTrackingCore/Private/OpenXRToLeapWrapper.cpp (hunk 2 above)
Comment (MaxPalmer-UH) [reply Source/UltraleapTrackingCore/Private/OpenXRToLeapWrapper.cpp:15]: I think internally OpenXR is the default?

**Comment by MaxPalmer-UH** [comment]
The first 10 or so commits are about getting full compatibility with LeapC across the API when using OpenXR. That's fine but was slightly unexpected for MetaHumans support.

File: Source/UltraleapTrackingCore/Private/OpenXRToLeapWrapper.cpp (hunk 2 above)
Comment (MaxPalmer-UH) [reply Source/UltraleapTrackingCore/Private/OpenXRToLeapWrapper.cpp:15]: So keep it.

File: UltraleapTracking.uplugin (hunk 1 above)
Comment (MaxPalmer-UH) [reply UltraleapTracking.uplugin:4]: I'm still seeing this as 4.6.2!

**Comment by MaxPalmer-UH** [comment]
I'm happy with these changes, just want to ensure we do a good regression pass before release.

**Pull request merged**